app.py

```diff
@@ -17,9 +17,18 @@ def home() -> str:
     return render_template('home.html')
 
 
+@app.route('/health')
+def healthcheck() -> tuple[str, int]:
+    return "200 OK", 200
+
+
 @app.route('/animals', methods=['GET'])
 def index() -> Response:
+    name_search = request.args.get('name')
     animals = Animal.query.all()
+    if name_search:
+        animals = [animal for animal in animals if name_search.lower() in animal.name.lower()]
```

Comment:
It will work, but better to use ORM for filtering directly from database.
As I understood you had problem with case and substring.
Also you can filter by part of string (substring) case insensetive, like explained here
https://stackoverflow.com/questions/4926757/sqlalchemy-query-where-a-column-contains-a-substring